stdlib/public/core/RangeReplaceableCollectionType.swift

```diff
@@ -446,10 +446,10 @@ public func +<
     S : SequenceType
     where S.Generator.Element == C.Generator.Element
 >(lhs: C, rhs: S) -> C {
-  var lhs = lhs
-  // FIXME: what if lhs is a reference type?  This will mutate it.
-  lhs.appendContentsOf(rhs)
-  return lhs
+  var result = lhs is AnyObject ? C(lhs) : lhs
```

Comment:
The library has to be not just correct, it has to be efficient.  Given that we know that we have other issues with reference-typed collections, I'm inclined to defer fixing this issue until we have tests that show that reference-typed collections work reasonably well in general.

I'm pretty sure that a lot of other algorithms rely on collections being value types, it is just that this particular piece of code happens to have a FIXME.  Once we know what the general issues are, we can start addressing them in a principled way, possibly adjusting the collections API along the way, to make it less error-prone to handle reference typed collections.